### PR TITLE
Adding keymaps for Niu Mini and BDN9

### PR DESCRIPTION
**Pull request by xplusplus**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keebio/bdn9/keymaps/codecoffeecode/config.h [hunk 1]
```diff
@@ -0,0 +1,11 @@
+
```
Comment (noroadsleft) [inline keyboards/keebio/bdn9/keymaps/codecoffeecode/config.h]:
```suggestion
#pragma once

```

File: keyboards/keebio/bdn9/keymaps/codecoffeecode/keymap.c [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        CTRL_END, _______, CTRL_HOME
+    ),
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    if (record->event.pressed) {
+        switch (keycode) {
+            case CTRL_HOME:
+                if (record->event.pressed) {
+                    SEND_STRING(SS_DOWN(X_LCTRL) SS_TAP(X_HOME) SS_UP(X_LCTRL));
```
Comment (noroadsleft) [inline keyboards/keebio/bdn9/keymaps/codecoffeecode/keymap.c]:
```suggestion
                    tap_code16(C(KC_HOME));
```

File: keyboards/keebio/bdn9/keymaps/codecoffeecode/keymap.c [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+                }
+                break;
+            case CTRL_END:
+                if (record->event.pressed) {
+                    SEND_STRING(SS_DOWN(X_LCTRL) SS_TAP(X_END) SS_UP(X_LCTRL));
```
Comment (noroadsleft) [inline keyboards/keebio/bdn9/keymaps/codecoffeecode/keymap.c]:
```suggestion
                    tap_code16(C(KC_END));
```

File: keyboards/keebio/bdn9/keymaps/codecoffeecode/keymap.c [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        CTRL_END, _______, CTRL_HOME
+    ),
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    if (record->event.pressed) {
+        switch (keycode) {
+            case CTRL_HOME:
+                if (record->event.pressed) {
+                    tap_code16(LCTL(KC_HOME));
```
Comment (drashna) [inline keyboards/keebio/bdn9/keymaps/codecoffeecode/keymap.c]: You don't need a macro for this.  You can just use `LCTL(KC_HOME)` in your layout.

File: keyboards/keebio/bdn9/keymaps/codecoffeecode/keymap.c [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+                }
+                break;
+            case CTRL_END:
+                if (record->event.pressed) {
+                    tap_code16(LCTL(KC_END));
```
Comment (drashna) [inline keyboards/keebio/bdn9/keymaps/codecoffeecode/keymap.c]: As above.

File: keyboards/keebio/bdn9/keymaps/codecoffeecode/keymap.c (hunk 4 above)
Comment (xplusplus) [reply keyboards/keebio/bdn9/keymaps/codecoffeecode/keymap.c]: Ah, I did not realize that. Thanks!

**Comment by noroadsleft** [comment]
Looks good, thanks!

**Pull request merged**